### ISSUE TITLE
add a test which runs all scripts in examples/ to check basic failure modes

### ISSUE THREAD

**Issue by dustinvtran** [blei-lab/edward#224]
This test shouldn't check that we recover good inferences or anything like that. It should simply check obvious failure modes, such as if the code immediately errors when you try to run it.

**Comment by akucukelbir** [comment]
After chatting with @dustinvtran we decided that tests should be carefully designed and that we would eventually curate the list of examples to be useful to newcomers.

**Comment by akucukelbir** [comment]
@dustinvtran do we still want to do this, based on some of our recent discussions?

**Comment by dustinvtran** [comment]
I think it would be good to test the notebooks in `docs/notebooks`; also see https://github.com/GPflow/GPflow/issues/373 and GPFlow's implementation (https://github.com/GPflow/GPflow/pull/374). As in their implementation, we can give each notebook a timeout option to avoid very long testing with notebooks such as `gan.ipynb`.

For the example scripts, I'm not sure. Your call!

**Comment by akucukelbir** [comment]
cool! looking into it now.

let's test the notebooks first. i'm not sure what'll happen to the examples in the near future.

**Comment by reinheimer** [comment]
There is a problem exactly with the GAN example in Jupyter. The final lines of the 
[complete trace](https://github.com/blei-lab/edward/files/865496/gan.zip):
```
...
InternalError (see above for traceback): Blas SGEMM launch failed : a.shape=(128, 100), b.shape=(100, 128), m=128, n=128, k=100
	 [[Node: Gen/fully_connected/MatMul = MatMul[T=DT_FLOAT, transpose_a=false, transpose_b=false, _device="/job:localhost/replica:0/task:0/gpu:0"](Gen/Uniform/sample/Reshape, Gen/fully_connected/weights/read)]]
```
Searching online, I just found this error associated with the running of more than one instance at the same time, which is not my case. Also, using the terminal, the example runs just fine. Should I submit an issue for edward repository?

I'm using tensorflow-gpu (1.0.1) and edward (1.2.4) in a virtual environment (venv in Python 3.4.2).

**Comment by dustinvtran** [comment]
Fixed in #542.
